File: exercise01-C/src/program.cpp
// Exercise 01-C: Count Duplicates

#include <iostream>
#include <string>
#include <unordered_set>

int main(int argc, char *argv[]) {
    int duplicates = 0;
    std::unordered_set<int> seen_numbers;

    int n;
    while (std::cin >> n)
        if (!seen_numbers.insert(n).second) duplicates++;

    std::cout << duplicates << '\n';

    return 0;
}

// vim: set sts=4 sw=4 ts=8 ft=cpp:

File: exercise01-C/src/program.py
#!/usr/bin/env python3

# Exercise 01-C: Count Duplicates

import sys


def main():
    duplicates = 0
    seen_numbers = set()

    for line in sys.stdin:
        if (n := int(line)) in seen_numbers:
            duplicates += 1
        else:
            seen_numbers.add(n)

    print(duplicates)


if __name__ == "__main__":
    main()

# vim: set sts=4 sw=4 ts=8 expandtab ft=python:
